SwiftGit2/Repository.swift

```diff
@@ -124,6 +124,8 @@ final public class Repository {
 	///
 	/// Returns a `Result` with a `Repository` or an error.
 	class public func at(_ url: URL) -> Result<Repository, NSError> {
+		SwiftGit2Init
```

Comment:
Slight nit-picking: While this does work, my opinion is that while injecting side-effects into global variable initializer is reasonable, but the way it's used looks a bit unnatural. Upon first glance one could not easily see that this is reading from the `SwiftGit2Init` variable, thereby executing `git_libgit2_init` exactly once.

How about:

```swift
// In Libgit2.swift, or SwiftGit2.swift
private var gitInitialized: Int = ({
	return Int(git_libgit2_init())
})()

func ensureGitInitialized() {
	precondition(gitInitialized >= 0)
}

// In places we want to use git, such as Repository.swift
ensureGitInitialized()
// ... start using git
```

Reply:
Addressed in commit https://github.com/SwiftGit2/SwiftGit2/pull/138/commits/5a28aea51a610f3e2222bfe0724e76bbd8018b07